src/redux/wallets.js

```diff
@@ -0,0 +1,112 @@
+import { get } from 'lodash';
+import {
+  generateAccount,
+  getAllWallets,
+  getSelectedWallet,
+  publicAccessControlOptions,
+  saveAddress,
+  saveAllWallets,
+  setSelectedWallet,
+} from '../model/wallet';
+import { colors } from '../styles';
+
+// -- Constants --------------------------------------- //
+const WALLETS_UPDATE = 'wallets/ALL_WALLETS_UPDATE';
+const WALLETS_LOAD = 'wallets/ALL_WALLETS_LOAD';
+const WALLETS_SET_SELECTED = 'wallets/SET_SELECTED';
+const WALLETS_ADDED_ACCOUNT = 'wallets/WALLETS_ADDED_ACCOUNT';
+
+// -- Actions ---------------------------------------- //
+export const walletsLoadState = () => async dispatch => {
+  try {
+    const { wallets } = await getAllWallets();
+    const selected = await getSelectedWallet();
+
+    dispatch({
+      payload: {
+        selected: get(selected, 'wallet', undefined),
+        wallets,
+      },
+      type: WALLETS_LOAD,
+    });
+    // eslint-disable-next-line no-empty
+  } catch (error) {}
+};
+
+export const walletsUpdate = wallets => dispatch => {
+  saveAllWallets(wallets);
+  dispatch({
+    payload: wallets,
+    type: WALLETS_UPDATE,
+  });
+};
+
+export const walletsSetSelected = wallet => dispatch => {
+  setSelectedWallet(wallet);
+  dispatch({
+    payload: wallet,
+    type: WALLETS_SET_SELECTED,
+  });
+};
+export const addressSetSelected = address => () => {
+  saveAddress(address, publicAccessControlOptions);
+};
+
+export const createAccountForWallet = id => async (dispatch, getState) => {
+  const { wallets } = { ...getState().wallets };
```

Comment:
This line doesn't need the destructuring as it is taken care of in the next line!

Reply:
oops! forgot to change that